modules/sc-mesh-secure-deployment/src/nats/src/comms_service_refresh.py

```diff
@@ -0,0 +1,59 @@
+"""
+Refresher class for the comms service registration
+"""
+import logging
+import threading
+import time
+import subprocess
+
+class CommsServiceRefresh:
+    """
+    Comms service publisher class
+    """
+    DNS_SERVICE_EVENT_LOOP_TIMEOUT: int = 5
+
+    def __init__(self, __hostname, __logger: logging.Logger) -> None:
+
+        self.logger: logging.Logger = __logger.getChild("CommsServicePublisher")
+        self.logger.setLevel(logging.INFO)
+        self.__event: threading.Event = threading.Event()
+        self.__hostname: str = __hostname
+
+
+    def __refresh_hostname(self) -> None:
+        """
+        Refresh the hostname with avahi-resolve-host-name
+        """
+        try:
+            subprocess.run(["avahi-resolve-host-name", self.__hostname + ".local"],
+                           check=True,
+                           stdout=subprocess.PIPE,
+                           stderr=subprocess.PIPE)
+        except subprocess.CalledProcessError:
+            self.logger.exception("Error refreshing hostname")
+
+    def dns_service_refresh(self) -> None:
+        """
+        Register and re-announce service periodically
+        """
+        self.logger.info("Refresh start")
+        while not self.__event.is_set():
+            time.sleep(self.DNS_SERVICE_EVENT_LOOP_TIMEOUT)
+            self.__refresh_hostname()
+        self.logger.info("Refresh stopped")
+
+
+    def shutdown_service(self) -> None:
+        """
+        Shutdown the service
+        """
+        self.__event.set()
+
+if __name__ == "__main__":
+    logging.basicConfig(level=logging.INFO)
+    logger = logging.getLogger("CommsServiceRefresh")
+    comms_service_refresh = CommsServiceRefresh("mika-battlestation", logger)
```

Comment:
Maybe you can change it to some other name than "mika-battlestation" :)

Reply:
😂 that was for personal testing purposes and actually I removed it in mdmagent repo.  I will remove the main function.

Reply:
[9e81465](https://github.com/tiiuae/mesh_com/pull/485/commits/9e81465dd3dfea8f88ddbf950b5d85c1ea3cfa78)